### DISCUSSION TITLE
Prisma explicit many to many field already defined

### DISCUSSION THREAD

**Discussion by alienbuild** [prisma/prisma#16435] [Q&A] [answered]
I have comics which has many writers, artists and characters and i'm trying to join everything together in prisma, but struggling to understand what I'm doing wrong.

Comic schema:
```
model comics {
  uniqueCoverId             String                    @id
  name                      String
  artists                   comic_contributors[]   @relation("artists")
  writers                   comic_contributors[]   @relation("writers")
}
```

Artists schema
```
model comic_artists {
  id                       String                    @id @default(uuid())
  name                     String?
  comics                   comic_contributors[]
}
```

Writers schema
```
model comic_writers {
  id     String                    @id @default(uuid())
  name   String?
  comics comic_contributors[]
}
```

Intermediary table
```
model comic_contributors {
  writer_id String              @unique
  writer    comic_writers? @relation("writers", fields: [writer_id], references: [id])

  artist_id String              @unique
  artist    comic_artists? @relation("artists", fields: [artist_id], references: [id])

  @@id([writer_id, artist_id])
}
```

When I try to format prisma it returns the errors

```
error: Field "comics" is already defined on model "comic_contributors".
comicsUniqueCoverId String?
comics              comics?        @relation(fields: [comicsUniqueCoverId], references: [uniqueCoverId])


error: Field "comicsUniqueCoverId" is already defined on model "comic_contributors".
comics              comics?        @relation(fields: [comicsUniqueCoverId], references: [uniqueCoverId])
comicsUniqueCoverId String?

```

Any suggestions on what I'm doing wrong please?

**Comment by alienbuild** [comment, marked as answer]
Fixed by switching db engine to innoDB

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
